java/java.hints/src/org/netbeans/modules/java/hints/infrastructure/JavaErrorProvider.java

```diff
@@ -113,7 +116,13 @@ public void run(ResultIterator it) throws Exception {
                                 if (disabled.size() != Severity.values().length) {
                                     AtomicBoolean cancel = new AtomicBoolean();
                                     context.registerCancelCallback(() -> cancel.set(true));
-                                    result.addAll(convert2Diagnostic(context.errorKind(), new HintsInvoker(HintsSettings.getGlobalSettings(), context.getOffset(), cancel).computeHints(cc), ed -> !disabled.contains(ed.getSeverity())));
+                                    
+                                    if (context.gethintsConfigFile() != null) {
```

Comment:
I think we can avoid some duplication here. Like:
- create `HintsSettings settings` variable outside of the if, fill it with `HintsSettings.createPreferencesBasedHintsSettings(hintSettings, true, null)` in one branch, and with `HintsSettings.getGlobalSettings()` in the other. The `result.addAll...` can then be outside of the if, shared by both codepaths.